server/base/src/main/java/org/apache/accumulo/server/fs/VolumeUtil.java

```diff
@@ -159,14 +160,14 @@ public static TabletFiles updateTabletVolumes(ServerContext context, ZooLock zoo
       }
     }
 
-    for (Entry<FileRef,DataFileValue> entry : tabletFiles.datafiles.entrySet()) {
-      String metaPath = entry.getKey().meta().toString();
+    for (Entry<TabletFile,DataFileValue> entry : tabletFiles.datafiles.entrySet()) {
+      String metaPath = entry.getKey().getNormalizedPath();
       Path switchedPath = switchVolume(metaPath, FileType.TABLE, replacements);
       if (switchedPath != null) {
         filesToRemove.add(entry.getKey());
-        FileRef switchedRef = new FileRef(switchedPath.toString(), switchedPath);
-        filesToAdd.put(switchedRef, entry.getValue());
-        ret.datafiles.put(switchedRef, entry.getValue());
+        TabletFile switchedFile = new TabletFile(switchedPath, switchedPath.toString());
```

Comment:
When I looked at this code I found myself wondering if any follow on code relies on the metadata entry for the TabletFile created here.  If so that could result in a silent bug (where we try to delete something in the metadata table that does not exists).  This makes me think it would be nice to be able to construct a TabletFile w/o a metadata entry and anything trying to get the metadata path would get an exception.  This would cause things to fail fast.   

Could possibly have a TabletFile constructor that only takes a path and when this constructor is used to create a the object, attempts to get the metadata entry result in an exception.

Reply:
That is a good point.  The volume replacement code (which is where this is used) will delete the original file but I don't see where it will write out the new path.  Is this done during compaction?  If so, then we will have problems if we throw an exception since the string won't ever exist in the metadata.

Reply:
> but I don't see where it will write out the new path. Is this done during compaction?

I am not sure, was there a specific class you were looking at?  I can take a look.


> If so, then we will have problems if we throw an exception since the string won't ever exist in the metadata.

It would be annoying if making TabletFile more strict causes code that is working for the wrong reasons to break, but it would be nice to cause code that is not working but failing silently to break.

Reply:
> > but I don't see where it will write out the new path. Is this done during compaction?
> 
> I am not sure, was there a specific class you were looking at? I can take a look.

Just this method, ```VolumeUtil.updateTabletVolumes()``` and the only place its called, [Tablet](https://github.com/apache/accumulo/blob/8ea08f8ef59cd199ec23d414d8dbc89be7eb1667/server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java#L342).

Reply:
I was looking at the code and it seems like there may be an existing bug.  I opened #1501